### DISCUSSION TITLE
Intensity changes on sh1106 with D1 Mini(ESP8266)

### DISCUSSION THREAD

**Discussion by jandemna** [olikraus/u8g2#1466] [Q&A]
Hi, I am Jan from NL
I like programming Arduinos and have attached to my D1 Mini an 1.3 inch Oled display for displaying time, weather and KAKU control.
The OLED is sh1106 SPI and works fine. Ug82 gives me nice fonts and symbols!
Problem I have is that the display slowly changes intensity from low to very bright. Cycle takes about 6s. Sometimes after new power up this not the case, but very often it is unfortunately. Looked at low level stuff already, but I am not an expert there. To me it look like "something" is not initialized and likes to come on different after power up. Any ideas welcome!
![2021-04-18 12 14 32](https://user-images.githubusercontent.com/34832939/115154440-2fe6ab80-a07b-11eb-9af8-188972591fc5.jpg)

**Comment by olikraus** [comment]
Looks like a hardware problem. U8g2 just sends the init sequence once, There is no other brightness change by u8g2 (unless you use the set contrast command).
I would suggest to resolder the parts on the back side. I mean: just hold your solder iron for 2-3 seconds to each solder area, melt the solder a little bit and check whether this improves the situation.

**Comment by jandemna** [comment]
OK, thought also might be hardware problem. Resoldered, but no improvement. Once in so many powers on I see the problem.
So as a next step I added u8g2.setContrast(value) to my program and made the value remote settable by web page. Only executes the setContast when value not zero.
The intensity of the display changes when I adjust the value between 1 and 127. 
Now when the intensity problem as described happens after power up, changing the value from setContrast does not have an effect. Display continues to go dark/brighter. Display works further ok.
THEN I looked at my test hardware and noticed that I had connected the display reset to Vcc directly. (because D3 used for temp sensor). 
U8G2_SH1106_128X64_NONAME_F_4W_HW_SPI u8g2(U8G2_R0, /* cs=*/ D8, /* dc=*/ D1, /* reset=*/ D3);
So I changed program that D3 could perform reset through u8g2 and I could not find any intensity changes any more after power on.
Conclusion: Reset through u8g2 matters. I also learned how to control the intensity with setContrast, which is a nice feature.
